en/02_Developer_Guides/02_Controllers/07_CMS_JSON_APIs.md

```diff
@@ -0,0 +1,154 @@
+
+Ensure the security of your endpoints by validating the token header using `SecurityToken::inst()->checkRequest($this->getRequest())` on relevant endpoints. If this check fails then call `$this->jsonError(400);` which is consistent with the 400 status code used in [`FormRequestHandler::httpSubmission()`](api:SilverStripe\Forms\FormRequestHandler::httpSubmission()) when the CSRF check fails when submitting data using `FormSchema`.
+
+## Passing values from PHP to JavaScript
+
+To transmit values from PHP to JavaScript, override `LeftAndMain::getClientConfig()` within your controller. Begin your method with `$clientConfig = parent::getClientConfig();` to ensure proper inheritance.
+
+Include any relevant links to endpoints in the client configuration. For example, add `'myEndpointUrl' => $this->Link('my-endpoint')`, where `my-endpoint` is specified in `private static array $url_handlers`.
+
+In JavaScript, access these values using `import Config from 'lib/Config';` and retrieve the endpoint URL using `Config.getSection('Full\ClassName\Of\MyController').anArrayKey.myEndpointUrl;`.
+
+## JavaScript AJAX requests
+
+Use the `backend` helper which is a wrapper around `fetch()` when making JavaScript requests. Import it using `import backend from 'lib/Backend';`.
+
+The `backend` helper is able to use a `catch()` block to handle 400-500 response codes, offering a more streamlined approach compared to using vanilla `fetch()`. It also provides handy shorthand methods such as `.get()` and `.post()`, for writing concise code.
+
+The following code will make a POST request to an endpoint passing JSON data in the request body.
+
+```js
+// MyComponent.js
+
+const endpoint = `${Config.getSection(section).someKey.someUrl}`;
+const data = { somekey: 123 };
+const headers = { 'X-SecurityID': Config.get('SecurityID') };
+backend.post(endpoint, data, headers)
+  .then(() => {
+    // handle 200-299 status code response here
+  })
+  .catch(() => {
+    // handle 400-500 status code response here
+  });
+```
+
+On the controller's endpoint method, retrieve the POST data using `$json = json_decode($this->getRequest()->getBody());`.
+
+## Unit testing
+
+Write unit tests with a subclass of [`FunctionalTest`](api:SilverStripe\Dev\FunctionalTest) instead of the regular [`SapphireTest`](api:SilverStripe\Dev\SapphireTest). This allows you to make HTTP requests to your endpoints and ensures comprehensive functional testing.
+
+Use `$this->get()` for endpoints designed to handle GET requests and `$this->post()` for those intended for POST requests. For non-GET/POST requests, such as DELETE, use `$this->mainSession->sendRequest('DELETE', $url, [], $headers);`.
+
+Make heavy use of dataProviders and pass in parameters such as `$expectedBody` and `$expectedStatusCode`. Ensure that every call to `jsonSuccess()` and `jsonError()` has at least one corresponding case in your dataProvider.
```

Comment:
```suggestion
See [functional testing](/developer_guides/testing/functional_testing/) for more information about setting up and running functional tests.
```

If that page doesn't already include this information, that's where it should be added.

Reply:
Updated. Functional testing docs already mention get() and post().